common/src/hash.rs

```diff
@@ -192,3 +178,19 @@ pub fn lcg_urandom(mut x: u32, buf: &mut [u8]) {
         *b = ((x >> 16) & 0xff) as u8;
     }
 }
+
+pub fn hash_object_id_raw(p: usize) -> PyHash {
+    // TODO: Use commented logic when below issue resolved.
+    // Ref: https://github.com/RustPython/RustPython/pull/3951#issuecomment-1193108966
+
+    // let mut y = p;
+    // /* bottom 3 or 4 bits are likely to be 0; rotate y by 4 to avoid
+    //    excessive hash collisions for dicts and sets */
+    // y = (y >> 4) | (y << (8 * std::mem::size_of::<usize>() - 4));
+    // y as PyHash
```

Comment:
Rust have [rotate_right](https://doc.rust-lang.org/std/primitive.u64.html#method.rotate_right) for this function